### DISCUSSION TITLE
Enhancements to group function

### DISCUSSION THREAD

**Discussion by AndyMcProducer** [jamulussoftware/jamulus#1049] [UI Related]
The grp option to be extended to have multiple groups with Solo and Mute Buttons for Groups, also namable.
This would help with live jams with many people, a waiting room would be great too. :P
Hiya great work all.

**Comment by gilgongo** [comment]
Hi Andy - nice idea! Can you edit the title of this ticket to be "Enhancements to group function" so that it doesn't get lost in the list? Thanks.

**Comment by corrados** [comment]
> This would help with live jams with many people

What do you mean by that? Do you mean the usecase that you want to live stream a session to YouTube and the one who mixes the audio signal for the stream could benefit from your enhancements?

**Comment by AndyMcProducer** [comment]
> > This would help with live jams with many people
> 
> What do you mean by that? Do you mean the usecase that you want to live stream a session to YouTube and the one who mixes the audio signal for the stream could benefit from your enhancements?

Yes Corrados, it would allow for better management of big servers with many people. Mainly for handling many artist for listeners. But even if not streamed live it would allow an admin to split up a jam so they could allow all some time to participate.

Could I suggest if a waiting room was made that maybe it should just be a basic chat and not a full connection, so a separate addition. Reason why would be to keep bandwidth down, stop pings rising with more connections so only minimal data was used and not to effect jams. I know a waiting room would need adjustment to server and client but it would give people what they need.

**Comment by corrados** [comment]
> Mainly for handling many artist for listeners

Then this would be a better solution:
https://github.com/corrados/jamulus/issues/146
Then you can use your DAW to do all these things.

> Could I suggest if a waiting room was made

Simon with his Jamulus World Jam on Facebook uses a separate Jamulus server for that. That works perfectly. I recommend to you to join his Facebook group.

**Comment by AndyMcProducer** [comment]
> > Mainly for handling many artist for listeners
> 
> Then this would be a better solution:
> #146
> Then you can use your DAW to do all these things.
> 
> > Could I suggest if a waiting room was made
> 
> Simon with his Jamulus World Jam on Facebook uses a separate Jamulus server for that. That works perfectly. I recommend to you to join his Facebook group.

That would be awesome addition but wasn't how I meant. Would be better being a vst3 to do that as you then have option to route multiple out and in's.

The idea would be to manage players via name and group so disconnects and reconnects wouldn't matter. Pushing to a daw would need that person on the exact same output if they disconnected and reconnected. Else it would mean every time someone dropped they'd need to be routed and setup every time. I meant more being able to simply group say a band, and with a click solo them or switch group so that audio of that group only. Even if not a waiting room the ability to have more than one group and a way to solo each group would be handy. I guess if for just jamming then no need for any options like these.

**Comment by corrados** [comment]
> I meant more being able to simply group say a band, and with a click solo them or switch group so that audio of that group only.

The current behavoir of the group is that the faders are synchronized. In your example, I assume, you do not want that to happen. You only want, e.g., the mute buttons being synchronized and not the faders, right?

**Comment by AndyMcProducer** [comment]
> > I meant more being able to simply group say a band, and with a click solo them or switch group so that audio of that group only.
> 
> The current behavior of the group is that the faders are synchronized. In your example, I assume, you do not want that to happen. You only want, e.g., the mute buttons being synchronized and not the faders, right?

Well having say 6 groups with faders or have them added on demand. Then a Fader can control all other volumes in that group. I say that way else you need to start routing audio but then that would be an advantage to other things later. 
Mute and Solo would simply be a group faders solo or mute that controlled channels attached to the group, however if routed then only 1 fader would need to solo to solo a group and if needed a muted channel associated with that group could be muted if needed. IF it was a set number of groups, say 4 or 6 then there could be a GRP 1, GRP 2 and so on buttons running down vertically on every channel, so selecting a group would send it to that group, so if someone needed to be in 2 groups as they say was playing in 2 bands live then they could be added to both. If no grp buttons selected then routes like normal.

Sometimes life throws us a curved ball, it might not be where we wanted to head but it can leave to much more and give us a good surprise.

**Comment by kezzy1966** [comment]
Just making sure I understand this please. Would an example of this Group Enhancement be having say all the guitars in one group and say all drummers in another ?

If so that is great as it could be used for choirs to have Tenors, Bass, Soprano, Alto separated.

**Comment by AndyMcProducer** [comment]
@corrados can't see how to message you on this so just delete when you've got the link, A friend posted it else where, tbh i haven't had a chance to look at it yet. Might be no good for Jamulus but might be worth reading anyway.
https://improbable.io/blog/kcp-a-new-low-latency-secure-network-stack

**Comment by gilgongo** [comment]
Hi all - until we get a clear picture of what needs to be done in terms of an item on the backlog, I'll move this to a discussion f that's OK.
